lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -132,15 +135,28 @@ export class PublicClientApplication implements IPublicClientApplication {
      * - if true, performs logic to cache and navigate
      * - if false, handles hash string and parses response
      */
-    private async handleRedirectResponse(): Promise<AuthenticationResult> {
+    private async handleRedirectResponse(): Promise<AuthenticationResult | null> {
         // Get current location hash from window or cache.
         const { location: { hash } } = window;
-        const cachedHash = this.browserStorage.getItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.URL_HASH), CacheSchemaType.TEMPORARY) as string;
         const isResponseHash = UrlString.hashContainsKnownProperties(hash);
-        const loginRequestUrl = this.browserStorage.getItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.ORIGIN_URI), CacheSchemaType.TEMPORARY) as string;
 
-        const currentUrlNormalized = UrlString.removeHashFromUrl(window.location.href);
+        // Check for interaction type
+        if (isResponseHash) {
+            // Deserialize hash fragment response parameters.
+            const hashUrlString: UrlString = new UrlString(hash);
```

Comment:
Wouldn't you need to pass in the whole url here? Why not just call `StringUtils.queryStringToObject` on the hash instead?